flink-end-to-end-tests/flink-confluent-schema-registry/src/main/java/AvroSerializationConfluentSchema.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.apache.flink.api.common.serialization.SerializationSchema;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import tech.allegro.schema.json2avro.converter.JsonAvroConverter;
+
+/**
+ * The serialization schema for the Avro type.
+ */
+public class AvroSerializationConfluentSchema<T> implements SerializationSchema<T> {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String schemaRegistryUrl;
+
+	private final int identityMapCapacity;
+
+	private KafkaAvroSerializer kafkaAvroSerializer;
+
+	private String topicName;
+
+	private SchemaRegistryClient schemaRegistryClient;
+
+	private JsonAvroConverter jsonAvroConverter;
+
+	private final Class<T> avroType;
+
+	public AvroSerializationConfluentSchema(Class<T> avroType, String schemaRegistryUrl, String topicName) {
```

Comment:
The serialization format should not need to know about the `topicName` which is used here for the schema lookup. Better to use `KafkaAvroEncoder` which automatically retrieves the schema name.